src/modules-load/modules-load.c

```diff
@@ -163,34 +179,44 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 log_warning_errno(r, "Failed to parse kernel command line, ignoring: %m");
 
-        r = module_setup_context(&ctx);
-        if (r < 0)
-                return log_error_errno(r, "Failed to initialize libkmod context: %m");
+        int sockets[2];
+        const int max_tasks = MAX(MIN(sysconf(_SC_NPROCESSORS_ONLN) / 2, 10), 2);
+        pthread_t *threads = malloc(sizeof(pthread_t) * max_tasks);
+        if (socketpair(AF_UNIX, SOCK_SEQPACKET, 0, sockets) == -1)
+                log_error_errno(errno, "socketpair");
```

Comment:
please fail the tool if this fails, i.e. add `return` before th `log_error_errno(`.

also, we always use `< 0` checks for validating success syscall returns rather than `== -1`. Please stick to that rule.

Also, please always add SOCK_CLOEXEC, as a ganeral rule in our codebase.